### ISSUE TITLE
alt-browser fallback to "browser" for browserification of dependencies?

### ISSUE THREAD

**Issue by mvayngrib** [defunctzombie/node-browser-resolve#62]
this is a question. If A depends on B, and their package.json's look like:

A:
```json
    {
      ...
      "chromeapp": {
        "dgram": "chrome-dgram"
      },
      "dependencies": {
        "B": "^1.0.0"
      }
      ...
    }
```
B:
```json
    {
      ...
      "browser": {
        "url": "./browser-url.js"
      }
      ...
    }
```

when you issue:
```
    browserify --browser chromeapp A/main.js
```

should browserify fall back to the "browser" field when it browserifies B and needs to resolve "url" but doesn't find a "chromeapp" field? It seems like in most cases this would be desirable. What do you think?

**Comment by defunctzombie** [comment]
Good question. What is the current behavior? No fallback? Silent failure? Loud failure?

**Comment by mvayngrib** [comment]
the current behavior is that it'll look for "chromeapp" in the nested dependency (where only "browser" exists), not find it, and resolve to whatever file/module gets used in the regular node.js environment. So basically it ends up handing you non-browserified deps

**Comment by defunctzombie** [comment]
fixed in v1.9.0

Thanks!

**Comment by mvayngrib** [comment]
cool, thx for merging
